packages/bot-web-ui/src/components/dashboard/dashboard-component/load-bot-preview/__tests__/recent.spec.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { render, screen } from '@testing-library/react';
+import { isMobile } from '@deriv/shared';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore, useDBotStore } from 'Stores/useDBotStore';
+import RecentComponent from '../recent';
+import DashboardStore from 'Stores/dashboard-store';
+import LoadModalStore from 'Stores/load-modal-store';
+import SaveModalStore from 'Stores/save-modal-store';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+}));
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({}));
+
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+
+const dashboard_strategy = {
+    name: '',
+    xml: '',
+    save_type: '',
+    timestamp: 1,
+};
+
+const dashboard_strategies = [
+    { ...dashboard_strategy, name: 'Strategy1', id: '1' },
+    { ...dashboard_strategy, name: 'Strategy2', id: '2' },
+];
+
+describe('RecentComponent', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeAll(() => {
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        mock_DBot_store = {
+            ...mock_DBot_store,
+            dashboard: {
+                ...mock_DBot_store.dashboard,
+                setPreviewOnPopup: jest.fn(),
+                setOpenSettings: jest.fn(),
+                setStrategySaveType: jest.fn(),
+                is_dark_mode: false,
+                initInfoPanel: jest.fn(),
+            } as DashboardStore,
+            load_modal: {
+                ...mock_DBot_store.load_modal,
+                loadFileFromLocal: jest.fn(),
+                toggleLoadModal: jest.fn(),
+                setLoadedLocalFile: jest.fn(),
+                preview_workspace: jest.fn() as unknown,
+                selected_strategy: jest.fn() as unknown,
+                tab_name: 'google_tab',
+                setDashboardStrategies: jest.fn(),
+                getDashboardStrategies: jest.fn(),
+                onDriveOpen: jest.fn(),
+                dashboard_strategies,
+                getRecentFileIcon: jest.fn(),
+                getSaveType: jest.fn(),
+            } as LoadModalStore,
+            save_modal: {
+                ...mock_DBot_store.save_modal,
+                toggleSaveModal: jest.fn(),
+                updateBotName: jest.fn(),
+                onConfirmSave: jest.fn(),
+                onDriveConnect: jest.fn(),
+            } as SaveModalStore,
+        };
```

Comment:
Why do we need this @maryia-matskevich-deriv 
Instead of this can we just mock the blocky part like we did in the following tests?
- https://github.com/binary-com/deriv-app/blob/master/packages/bot-web-ui/src/components/flyout/__tests__/flyout-block.spec.tsx
- https://github.com/binary-com/deriv-app/blob/master/packages/bot-web-ui/src/components/quick-strategy/__tests__/form.spec.tsx

If we do it this way we don't need to mock all the methods, kindly let me know if you need any help

Reply:
refactored✅